main/main.go

```diff
@@ -162,55 +167,72 @@ func main() {
 		}),
 	)
 
-	http.Handle("/health", health.NewHandler(checker))
-
-	// start the health check server
-	go func() {
-		log.Fatalln(http.ListenAndServe(fmt.Sprintf(":%d", cfg.APIPort), nil))
-	}()
-
 	startMetricsServer(logger, gatherer, cfg.MetricsPort)
 
-	metrics, err := relayer.NewApplicationRelayerMetrics(registerer)
+	relayerMetrics, err := relayer.NewApplicationRelayerMetrics(registerer)
 	if err != nil {
-		logger.Error(
-			"Failed to create application relayer metrics",
-			zap.Error(err),
-		)
+		logger.Fatal("Failed to create application relayer metrics", zap.Error(err))
 		panic(err)
 	}
 
 	// Initialize message creator passed down to relayers for creating app requests.
 	messageCreator, err := message.NewCreator(logger, registerer, "message_creator", constants.DefaultNetworkCompressionType, constants.DefaultNetworkMaximumInboundTimeout)
 	if err != nil {
-		logger.Error(
-			"Failed to create message creator",
-			zap.Error(err),
-		)
+		logger.Fatal("Failed to create message creator", zap.Error(err))
 		panic(err)
 	}
 
 	// Initialize the database
 	db, err := database.NewDatabase(logger, &cfg)
 	if err != nil {
-		logger.Error(
-			"Failed to create database",
-			zap.Error(err),
-		)
+		logger.Fatal("Failed to create database", zap.Error(err))
 		panic(err)
 	}
 
 	// Initialize the global write ticker
 	ticker := utils.NewTicker(cfg.DBWriteIntervalSeconds)
 	go ticker.Run()
 
+	messageHandlerFactories, err := createMessageHandlerFactories(logger, &cfg)
+	if err != nil {
+		logger.Fatal("Failed to create Message Handler Factories", zap.Error(err))
+		panic(err)
+	}
+
+	applicationRelayers, minHeights, err := createApplicationRelayers(
+		context.Background(),
+		logger,
+		relayerMetrics,
+		db,
+		ticker,
+		network,
+		messageCreator,
+		&cfg,
+		sourceClients,
+		destinationClients,
+	)
+	if err != nil {
+		logger.Fatal("Failed to create Application Relayers", zap.Error(err))
+		panic(err)
+	}
+	relayer.SetMessageCoordinator(logger, messageHandlerFactories, applicationRelayers, sourceClients)
+
+	// Initialize the API after the message coordinator is set
+	http.Handle("/health", health.NewHandler(checker))
+	http.Handle(relayer.RelayMessageApiPath, relayer.RelayMessageAPIHandler())
+
+	// start the health check server
+	go func() {
+		log.Fatalln(http.ListenAndServe(fmt.Sprintf(":%d", cfg.APIPort), nil))
+	}()
```

Comment:
I'm not sure exactly what you mean here. Do you mean instantiating a new `ServeMux` instead of using the default one for everything here? Or encapsulating all this logic in a separate package?

Reply:
Sorry, my original comment was pretty unspecific. Now that we have two distinct endpoints, it might make sense to put them and their handlers into their own package, rather than the current organization with one handler in package `main`, and another in package `relayer`. I don't feel super strongly though, especially if it is a significant rework or breaks the existing dependency graph.